src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +737,240 @@ impl<
+    /// It returns true if the veto succeeded, and false if it failed.
+    pub fn read_veto_state(
+        &mut self,
+        rc_cycle_of_veto: u64,
+        proposed_exit_rc: u64,
+        stacks_block_id: &StacksBlockId,
+    ) -> Result<bool, Error> {
+        let reward_cycle_length = self.burnchain.pox_constants.reward_cycle_length;
+        let veto_percent_threshold = self
+            .burnchain
+            .exit_contract_constants
+            .veto_confirmation_percent;
+        let is_mainnet = self.burnchain.is_mainnet();
+        self.chain_state_db
+            .with_read_only_clarity_tx(&self.sortition_db.index_conn(), &stacks_block_id, |conn| {
+                conn.with_clarity_db_readonly(|db| {
+                    // TODO (#3034): get correct contract ID once contract is published
+                    let exit_at_rc_contract = boot_code_id("exit-at-rc", is_mainnet);
+
+                    // from map rc-proposal-vetoes, use key pair (proposed_rc, curr_rc) to get the # of vetos
+                    let entry = db
+                        .fetch_entry_unknown_descriptor(
+                            &exit_at_rc_contract,
+                            "rc-proposal-vetoes",
+                            &Value::from(
+                                TupleData::from_data(vec![
+                                    ("proposed-rc".into(), Value::UInt(proposed_exit_rc as u128)),
+                                    ("curr-rc".into(), Value::UInt(rc_cycle_of_veto as u128)),
+                                ])
+                                .expect("BUG: failed to construct simple tuple"),
+                            ),
+                        )
+                        .expect("BUG: Failed querying rc-proposal-vetoes")
+                        .expect_optional();
+
+                    let num_vetos = match entry {
+                        Some(val) => {
+                            let tuple_data = val.expect_tuple();
+                            tuple_data
+                                .get("vetoes")
+                                .expect("BUG: malformed cost proposal tuple")
+                                .clone()
+                                .expect_u128()
+                        }
+                        None => 0,
+                    };
+
+                    println!(
+                        "RCPR: in veto check, proposed_exit: {}, num vetos: {:?}",
+                        proposed_exit_rc, num_vetos
+                    );
+                    // Check if the percent veto crosses the minimum threshold
+                    let percent_veto = num_vetos * 100 / (reward_cycle_length as u128);
+
+                    Ok(percent_veto >= (veto_percent_threshold as u128))
+                })
+            })
+            .ok_or(Error::DBError(NotFoundError))?
+    }
+
+    /// Returns map of exit proposals to the number of votes for each, as well as the sum total of all
+    /// votes.
```

Comment:
Can you update this comment to describe the range of key values for the resulting `BTreeMap`?  I.e. the keys are reward cycle IDs that fall into a particular range relative to `rc_cycle_of_vote`?